evaluation_plans/osps/quality/steps.go

```diff
@@ -188,3 +189,151 @@ func hasOneOrMoreStatusChecks(payloadData interface{}, _ map[string]*layer4.Chan
+	if message != "" {
+		return layer4.Unknown, message
+	}
+
+	// Validate required fields
+	if data.Repository.Name == "" || data.Repository.DefaultBranchRef.Name == "" ||
+		data.Repository.DefaultBranchRef.Target.OID == "" {
+		return layer4.Unknown, "Missing required repository data"
+	}
+
+	// Check dependency manifests
+	result, message = verifyDependencyManifests(data)
+	if result != layer4.Passed {
+		return result, message
+	}
+
+	return layer4.Passed, "Dependency management files and SBOM requirements met"
+}
+
+type DependencyManifest struct {
+	Filename    string
+	LockFile    string
+	Language    string
+	IsMandatory bool
+}
+
+// Known dependency management files for various languages
+var knownManifests = []DependencyManifest{
+	{"go.mod", "go.sum", "Go", true},
+	{"package.json", "package-lock.json", "JavaScript/Node.js", true},
+	{"pom.xml", "", "Java (Maven)", false},
+	{"build.gradle", "gradle.lockfile", "Java (Gradle)", false},
+	{"requirements.txt", "", "Python", false},
+	{"Pipfile", "Pipfile.lock", "Python (Pipenv)", true},
+	{"poetry.lock", "pyproject.toml", "Python (Poetry)", true},
+	{"Gemfile", "Gemfile.lock", "Ruby", true},
+	{"composer.json", "composer.lock", "PHP", true},
+	{"Cargo.toml", "Cargo.lock", "Rust", true},
+	{"*.csproj", "packages.lock.json", ".NET", false},
+	{"mix.exs", "mix.lock", "Elixir", true},
+}
+
+// ManifestResult stores the validation result for a manifest
+type ManifestResult struct {
+	Found     bool
+	LockFound bool
+	Language  string
+	HasDeps   bool
+	Manifest  string
+}
+
+func isManifestFile(filename string, pattern string) bool {
+	if strings.Contains(pattern, "*") {
+		matched, _ := filepath.Match(pattern, filename)
+		return matched
+	}
+	return filename == pattern
+}
+
+func verifyDependencyManifests(payloadData interface{}) (layer4.Result, string) {
+	data, message := reusable_steps.VerifyPayload(payloadData)
+	if message != "" {
+		return layer4.Unknown, message
+	}
+
+	if data.Repository.DefaultBranchRef.Target.Tree.Entries == nil {
+		return layer4.Unknown, "Repository tree entries not available"
+	}
+
+	entries := data.Repository.DefaultBranchRef.Target.Tree.Entries
+	manifests := data.Repository.DependencyGraphManifests.Nodes
+
+	foundManifests := make(map[string]ManifestResult)
+	manifestErrors := []string{}
+
+	// Check repository contents for dependency files
+	for _, entry := range entries {
+		for _, manifest := range knownManifests {
+			if isManifestFile(entry.Name, manifest.Filename) {
+				result := ManifestResult{
+					Found:    true,
+					Language: manifest.Language,
+					Manifest: entry.Name,
+				}
+
+				// Check for lock file if required
+				if manifest.LockFile != "" {
+					lockFound := false
+					for _, lockEntry := range entries {
+						if isManifestFile(lockEntry.Name, manifest.LockFile) {
+							result.LockFound = true
+							lockFound = true
+							break
+						}
+					}
+
+					if manifest.IsMandatory && !lockFound {
```

Comment:
https://baseline.openssf.org/versions/2025-02-25#osps-br-05---all-build-and-release-pipelines-must-use-standardized-tooling-where-available-to-ingest-dependencies-at-build-time is the only controls reference to the usage of lockfiles.

I like that you introduced awareness of them in this change, but I wonder if that logic should be moved over the evaluations for the BR control family

Reply:
I believe lock files were intentionally omitted here, so I think y'all are on the right track